chia/wallet/wallet.py

```diff
@@ -168,7 +168,7 @@ def puzzle_for_pk(self, pubkey: G1Element) -> Program:
         return puzzle_for_pk(pubkey)
 
     def puzzle_hash_for_pk(self, pubkey: G1Element) -> bytes32:
-        return puzzle_hash_for_pk(pubkey)
+        return self.wallet_state_manager.decorator_manager.decorate(puzzle_for_pk(pubkey)).get_tree_hash()
```

Comment:
For a puzzle decorator, there are 4 pre-defined methods.
decorate: Modify the primitive puzzle in a specific way, usually wrapping it with an additional puzzle. In clawback, we don't change anything here since the coin is in a normal P2 puzzle.

decorate_target_puzhash: Modify the target puzzle hash based on the requirements. In clawback, we need to send coins to a merkle puzzle hash instead of a P2 puzzle hash.

decorate_memos: Modify the memo based on the requirements. In clawback, we need to use the merkle puzzle hash as the hint instead of the P2 puzzle hash.

solve: Create a solution for the decorated puzzle.